crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -7743,7 +7743,9 @@ impl<'db> TypeInferenceBuilder<'db> {
         message: std::fmt::Arguments,
     ) -> Type<'db> {
         if let Some(builder) = self.context.report_lint(&INVALID_TYPE_FORM, expression) {
-            builder.into_diagnostic(message);
+            let mut diag = builder.into_diagnostic(message);
+            diag.info("See the following page for a reference on valid type expressions:");
+            diag.info("https://typing.python.org/en/latest/spec/annotations.html#type-and-annotation-expressions");
```

Comment:
I remember @AlexWaygood saying that the spec is not supposed to be user facing, but this seems clearly helpful. Or are there any objections?

Reply:
I think this is perfect!